src/main/src/test/java/org/geoserver/util/EntityResolverProviderTest.java

```diff
@@ -0,0 +1,124 @@
+        // include defaults if allow list is empty
+        String allowed[] = EntityResolverProvider.entityResolutionAllowlist("");
+        assertNotNull("defaults for empty", allowed);
+        assertEquals(4, allowed.length);
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.W3C::equals));
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.INSPIRE::equals));
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.OGC1::equals));
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.OGC2::equals));
+    }
+
+    @Test
+    public void testAllowListUnrestriced() throws Exception {
+        String allowed[] = EntityResolverProvider.entityResolutionAllowlist("*");
+        assertNull("* for Unrestricted", allowed);
+
+        allowed = EntityResolverProvider.entityResolutionAllowlist(null);
+        assertNull("null for Unrestricted", allowed);
+    }
+
+    @Test
+    public void testAllowListDomains() throws Exception {
+        String allowed[] = EntityResolverProvider.entityResolutionAllowlist("how2map.com");
+        // confirm allowed includes the defaults
+        assertNotNull(allowed);
+        assertEquals(5, allowed.length);
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.W3C::equals));
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.INSPIRE::equals));
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.OGC1::equals));
+        assertTrue(Arrays.stream(allowed).anyMatch(AllowListEntityResolver.OGC2::equals));
+        // in addition to the provided domain
+        assertTrue(Arrays.stream(allowed).anyMatch("how2map.com"::equals));
+    }
+
+    @Test
+    public void testNoWildcard() throws Exception {
+        // AllowListEntityResolver uses '*' to allow all http content (null returned)
+        String everything[] =
+                EntityResolverProvider.entityResolutionAllowlist(
+                        AllowListEntityResolver.UNRESTRICTED);
+        assertNull("* allows everything", everything);
+
+        // but wild cards such as `foo*bar` are not supported, strings are quoted and not intended
+        // to be a RegEx
+        String allowed[] = EntityResolverProvider.entityResolutionAllowlist("foo*bar");
+        assertNotNull(allowed);
+        assertTrue(Arrays.stream(allowed).anyMatch("foo*bar"::equals));
+    }
+
+    /**
+     * Test behaviour of EntityResolveProvider in response to configuration.
+     *
+     * <p>EntityResolver returns {@code null} when the provided URI is *allowed*, Returns an
+     * Inputstream if the content is provided, or throws an Exception if the URI is not allowed.
+     */
+    @Test
+    public void testEntityResolverDefaultBehaviour() throws Exception {
+        EntityResolverProvider provider = new EntityResolverProvider(null);
+        provider.setEntityResolver(new AllowListEntityResolver(null));
```

Comment:
This test case looks like its providing null (meaning accept everything) - but the test case is testing a dis-allow.

Reply:
sigh, okay! hold on ...

Reply:
@davidblasby I have added some additional checks (they were indeed useful and found problems); so it is worth forward porting the additional checks.

GeoServer is mostly test cases...